src/execution/compiler/operator/seq_scan_translator.cpp

```diff
@@ -85,17 +85,20 @@ void SeqScanTranslator::GenerateFilterClauseFunctions(util::RegionVector<ast::Fu
                                                       bool seen_conjunction) {
   // The top-most disjunctions in the tree form separate clauses in the filter manager.
   if (!seen_conjunction && predicate->GetExpressionType() == parser::ExpressionType::CONJUNCTION_OR) {
-    std::vector<ast::Identifier> next_clause;
-    GenerateFilterClauseFunctions(decls, predicate->GetChild(0), &next_clause, false);
-    filters_.emplace_back(std::move(next_clause));
-    GenerateFilterClauseFunctions(decls, predicate->GetChild(1), curr_clause, false);
+    for (size_t idx = 0; idx < predicate->GetChildrenSize() - 1; ++idx) {
+      std::vector<ast::Identifier> next_clause;
+      GenerateFilterClauseFunctions(decls, predicate->GetChild(idx), &next_clause, false);
+      filters_.emplace_back(std::move(next_clause));
+    }
+    GenerateFilterClauseFunctions(decls, predicate->GetChild(predicate->GetChildrenSize() - 1), curr_clause, false);
```

Comment:
Can you add comments that briefly explain what is going on here? Maybe include a simple explanation for what happens with `a=1 OR b=2 OR c=3`? 

What is also the significance of `seen_conjunction` for entering into this block of code?

Reply:
Added some explaination for a=1 OR b=2 OR c=3 and seen_conjunction